statsd/uds.go

```diff
@@ -20,33 +25,90 @@ type udsWriter struct {
 }
 
 // newUDSWriter returns a pointer to a new udsWriter given a socket file path as addr.
-func newUDSWriter(addr string, writeTimeout time.Duration) (*udsWriter, error) {
-	udsAddr, err := net.ResolveUnixAddr("unixgram", addr)
-	if err != nil {
-		return nil, err
-	}
+func newUDSWriter(addr string, writeTimeout time.Duration, transport string) (*udsWriter, error) {
 	// Defer connection to first Write
-	writer := &udsWriter{addr: udsAddr, conn: nil, writeTimeout: writeTimeout}
+	writer := &udsWriter{addr: addr, transport: transport, conn: nil, writeTimeout: writeTimeout}
 	return writer, nil
 }
 
+// retryOnWriteErr returns true if we should retry writing after a write error
+func (w *udsWriter) retryOnWriteErr(err error) bool {
+	if err == nil {
+		return true
```

Comment:
I think this is incorrect.

Reply:
removed this branch

Reply:
I think it was ok to have it (I mean, what you tried to do with it `if err == nil { return false }`